www/activities/foods-meals-recipes/js/food-editor.js

```diff
@@ -124,6 +153,21 @@ app.FoodEditor = {
       app.FoodEditor.changeServing(app.FoodEditor.item, "portion", e.target.value);
     });
     app.FoodEditor.el.portion.addEventListener("change", (e) => {
+      // Force to only allow the given pattern as input
+      if (!app.FoodEditor.el.portion.validity.valid)
+      {
+        e.target.value = e.target.oldValue | 0;
+        return;
+      }
+
+      if (!/^\d+$/.test(e.target.value))
+      {
+        let newValue = evalInpEquation(e.target.value)
+        if (!isNaN(newValue))
+        {
+          e.target.value = newValue;
+        }
+      }
       if (e.target.oldValue == undefined && e.target.value != 0)
```

Comment:
The new code for parsing needs to be placed below this code block (instead of above) because otherwise this code here might never be reached due to the early return statement.